server/auth/types/jwt/jwt_auth.ts

```diff
@@ -175,6 +175,11 @@ export class JwtAuthentication extends AuthenticationType {
     );
   }
 
+  // JWT auth types should get expiry from the JWT, and not override this value
+  public async supportsKeepAlive(request: OpenSearchDashboardsRequest): Promise<boolean> {
+    return false;
```

Comment:
@cwperks I am not too familiar with JWT, but I think this should be false. However when investigating the code, it seems like the only setting of expiryTime is here: https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/jwt/jwt_auth.ts#L163, and it seems to coincide with the ttl, not the expiry from the JWT. Are we not handling this case properly, wdyt?

Reply:
Hmm, I guess we need to think of those separately.

1. expiryTime associated with the cookie
2. `exp` claim of the JWT

When the backend security plugin authenticates the request it will always use the `exp` claim in the token to determine if the token is valid or not. The expiryTime in the cookie would never be considered when determining if the request is authenticated.

IMO the expiryTime should take the `exp` claim from the token and if the `exp` claim does not exist then the token is valid indefinitely. I'll look at the history of expiryTime in the jwt_auth to see the reason this was added.

Reply:
@derek-ho I've thought about this some more and I think we should treat JWT similar to username and password except in the case where the `exp` in the JWT would be less than the session.ttl if keepalive is set to true.

Here's why:

1. **Indefinite or long-dated JWTs** - The JWT could have no `exp` claim at all (token with indefinite life) or it can have a very long dated `exp` like 99-yr into the future. In this case, we should treat the JWT like we would a username and password. Passwords are indefinite for internal users and only change if the user explicitly changes it. Likewise with username and password the user should be logged out at the end of a session and have to re-enter the jwt to login again.

2. **JWTs with expiry less than session length** - In this case, the JWT expiry should override the session expiry and the user should be logged out when the JWT becomes invalid

tldr; For JWT take the lesser of the `exp` claim in the JWT or the session length to determine the length of the session. After session expiry the user logs out and if the JWT is still valid they can opt to re-login by presenting the token again.

Reply:
Agree, changed to match this behavior, and added many test cases to verify this behavior. I tested it on my local and verified that even without this no data is being returned unexpectedly, so this is more just to be explicit than patching bad behavior.